tsl/src/bgw_policy/compression_api.c

```diff
@@ -201,6 +217,13 @@ policy_compression_add_internal(Oid user_rel_oid, Datum compress_after_datum,
 	hcache = ts_hypertable_cache_pin();
 	hypertable = validate_compress_chunks_hypertable(hcache, user_rel_oid, &is_cagg);
 
+	/* creation time usage not supported with caggs yet */
+	if (is_cagg && created_before != NULL)
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("cannot use \"compress_created_before\" with continuous aggregate \"%s\" ",
+						get_rel_name(user_rel_oid))));
+
```

Comment:
But we are supporting it with normal time? What is the reason for why we cannot support this for the materialized hypertable? (Actually, is it at all possible to create a continuous aggregate table with a non-time columns?)

Reply:
This is based on initial feedback from Fabrizio, where he said we will need more work with Caggs. So erroring out till we get around to that. AFAIK, it's possible to create CAggs with `integer_now` support for non-time columns.